port-chooser/src/main/java/org/terracotta/utilities/test/net/EphemeralPorts.java

```diff
@@ -0,0 +1,336 @@
+    if (Os.isHpux()) { return new SolarisAndHPUX(true).getRange(); }
+
+    throw new AssertionError("No support for this OS: " + Os.getOsName());
+  }
+
+  /**
+   * Execute a command and return the first response line.
+   * @param cmd the command to execute
+   * @return the first response line
+   * @throws IndexOutOfBoundsException if the command response is empty
+   * @throws IOException if an error is encountered while executing the command
+   */
+  private static String firstLine(String... cmd) throws IOException {
+    Shell.Result result = Shell.execute(Shell.Encoding.CHARSET, cmd);
+    if (result.exitCode() == 0) {
+      return result.lines().get(0);
+    } else {
+      throw new IOException(String.format("Command %s failed; rc=%d:%n    %s",
+          Arrays.toString(cmd), result.exitCode(), String.join("\n    ", result)));
+    }
+  }
+
+  /**
+   * Execute a command and return an {@code Iterable} with all response lines.
+   * @param cmd the command to execute
+   * @return an {@code Iterable} over all response lines
+   * @throws IOException if an error is encountered while executing the command
+   */
+  private static Iterable<String> allLines(String... cmd) throws IOException {
+    Shell.Result result = Shell.execute(Shell.Encoding.CHARSET, cmd);
+    if (result.exitCode() == 0) {
+      return result;
+    } else {
+      throw new IOException(String.format("Command %s failed; rc=%d:%n    %s",
+          Arrays.toString(cmd), result.exitCode(), String.join("\n    ", result)));
+    }
+  }
+
+  /**
+   * Execute a command and return a {@code Properties} instance composed from the output.
+   * @param cmd the command to execute
+   * @return a new {@code Properties} instance
+   * @throws IOException if an error is encountered while executing the command or loading
+   *        the {@code Properties} instance
+   */
+  private static Properties asProperties(String... cmd) throws IOException {
+    Shell.Result result = Shell.execute(Shell.Encoding.CHARSET, cmd);
+    if (result.exitCode() == 0) {
+      Properties props = new Properties();
+      try {
+        props.load(new StringReader(String.join(System.lineSeparator(), result)));
+      } catch (IOException e) {
+        throw new IOException(String.format("Failed to load Properties from %s:%n    %s",
+            Arrays.toString(cmd), String.join("\n    ", result)), e);
+      }
+      return props;
+    } else {
+      throw new IOException(String.format("Command %s failed; rc=%d:%n    %s",
+          Arrays.toString(cmd), result.exitCode(), String.join("\n    ", result)));
+    }
+  }
+
+  public static class Range {
+    private final int upper;
+    private final int lower;
+
+    private Range(int lower, int upper) {
+      this.lower = lower;
+      this.upper = upper;
+    }
+
+    /**
+     * The last port assigned to the ephemeral/dynamic port range.
+     * @return the upper bound of the ephemeral port range
+     */
+    public int getUpper() {
```

Comment:
It bothers my brain that this is an inclusive upper bound.

Reply:
Turns out that's the meaning up the upper bound that configurable systems use -- when they specify an upper bound as opposed to a count, that is.